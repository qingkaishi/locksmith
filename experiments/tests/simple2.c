#include <pthread.h>

pthread_mutex_t lock;
int shared;
int non_protected;

void *run(void *arg) {
	non_protected++;
  pthread_mutex_lock(&lock);
  shared++;
	pthread_mutex_unlock(&lock);
  return NULL;
}
int main() {
  pthread_t t1, t2;
  pthread_mutex_init(&lock, NULL);
  pthread_create(&t1, NULL, run, NULL);
  pthread_create(&t1, NULL, run, NULL);
  return 1;
}
