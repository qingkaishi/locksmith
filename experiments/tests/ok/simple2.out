************************* STARTING *************************

shared: &non_protected:simple2.c:5
        &shared:simple2.c:4
&shared:simple2.c:4 is protected by:
  concrete lock:simple2.c:16
  lock:simple2.c:3

Warning: Possible data race: &non_protected:simple2.c:5 is not protected!
 references:
  dereference of &non_protected:simple2.c:5 at simple2.c:8
    &non_protected:simple2.c:5
  locks acquired:
    <empty>
  in: main at simple2.c:14 


*************************** DONE ***************************
